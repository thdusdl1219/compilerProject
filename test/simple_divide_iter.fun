/*return: (Quotient,remainder)*/
fun divide_iter(num_num:<int,int>):<int,int> =
let dividend=ref (#0 num_num) in
let divisor=#1 num_num in
let Quotient=ref 0 in
	(while not ((!dividend)<divisor) do
	 (dividend:=!dividend-divisor;
	  Quotient:=!Quotient+1)
	;<!Quotient,!dividend>)
		
fun main(argc:int):int =
	let ret= divide_iter(<10000,17>) in
		printint(#0 ret);
		printint(#1 ret);0
