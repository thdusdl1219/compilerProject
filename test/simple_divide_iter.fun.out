[588, 4, 0]
